### ISSUE TITLE
Inline-def rule does not warn for deftest inside deftest

### ISSUE THREAD

**Issue by egs33** [clj-kondo/clj-kondo#2603]
- [x] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**version**
clj-kondo v2025.07.28

**platform**

Native (linux-amd64)

**editor**

CLI

**problem**

The inline-def rule does not trigger a warning when a deftest is defined inside another deftest.

**repro**

```repro.clj
(ns repro
  (:require
   [clojure.test :refer [deftest is]]))

(defn top-level []
  (defn inner [])) ; warn

(deftest top-test
  (def inner-def 1) ; warn
  (deftest inner-test ; not warn
    (is (= 1 (inc 0)))))

```

output
```
clj-kondo --lint repro.clj
repro.clj:6:3: warning: inline def
repro.clj:9:3: warning: inline def
linting took 17ms, errors: 0, warnings: 2
```

**expected behavior**
I expect the inline-def rule to warn for deftest defined inside another deftest, just as it does for def and defn inside non-top-level forms.
